AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/helper/IntentBuilder.java

```diff
@@ -135,21 +138,36 @@ public JInvocation getSuperPutExtraInvocation(TypeMirror elementType, JVar extra
 		if (elementType.getKind() == TypeKind.DECLARED) {
 			Elements elementUtils = environment.getProcessingEnvironment().getElementUtils();
 			TypeMirror parcelableType = elementUtils.getTypeElement(PARCELABLE).asType();
-			if (!typeUtils.isSubtype(elementType, parcelableType)) {
-				TypeMirror stringType = elementUtils.getTypeElement(STRING).asType();
-				if (!typeUtils.isSubtype(elementType, stringType)) {
-					extraParameterArg = cast(environment.getClasses().SERIALIZABLE, extraParameterArg);
-				}
-			} else {
+			if (typeUtils.isSubtype(elementType, parcelableType)) {
 				TypeMirror serializableType = elementUtils.getTypeElement(SERIALIZABLE).asType();
 				if (typeUtils.isSubtype(elementType, serializableType)) {
 					extraParameterArg = cast(environment.getClasses().PARCELABLE, extraParameterArg);
 				}
+			} else if (isAnnotatedWith(elementType, "org.parceler.Parcel")) {
+				extraParameterArg = environment.getCodeModel().ref("org.parceler.Parcels").staticInvoke("wrap").arg(extraParameterArg);
+			} else {
+				TypeMirror stringType = elementUtils.getTypeElement(STRING).asType();
+				if (!typeUtils.isSubtype(elementType, stringType)) {
+					extraParameterArg = cast(environment.getClasses().SERIALIZABLE, extraParameterArg);
+				}
 			}
 		}
 		return _super().invoke("extra").arg(extraKeyField).arg(extraParameterArg);
 	}
 
+	public boolean isAnnotatedWith(TypeMirror typeMirror, String qualifedName) {
```

Comment:
I'd pass the `AnnotationHelper` to the intent builder.

Add it to the constructor, then update the `ActivityIntentBuilder` and `ServiceIntentBuilder` to constructors and thier instantiations. where they are instantiated there is `annotationHelper` field from parent class available.

but i think you should do that in a separate commit before doing the work that adds the parceler support